Framework/Kernel/src/DateAndTimeHelpers.cpp

```diff
@@ -73,6 +75,26 @@ std::string verifyAndSanitizeISO8601(const std::string &date,
   return date;
 }
 
+/**
+ * @brief averageSorted Assuming that the vector is sorted, find the average
+ * time
+ */
+Types::Core::DateAndTime
+averageSorted(const std::vector<Types::Core::DateAndTime> &times) {
+  if (times.empty())
+    throw std::invalid_argument("Cannot find average of empty vector");
+
+  // to avoid overflow subtract the first time off from everything
+  // and find the average in between
+  const int64_t first = times.begin()->totalNanoseconds();
```

Comment:
You could also use `times.front().totalNanoseconds()` to avoid reference semantics.